### PR TITLE
Make Stats V4 Default and remove stats store availability banner

### LINKED ISSUE #2539: Make v4 stats the default
1. If we know for sure the new stats endpoint is available, then default to the new stats UI instead of displaying a banner message.
2. Remove the beta feature option for using the improved stats since it will be enabled by default.
3. ~Site does not have v4 wc-analytics namespace available OR Woo version is less than 4.0 - inform user that old stats will eventually go away, upgrade to 4.0 or install WCAdmin.~ This was already [added in this PR](https://github.com/woocommerce/woocommerce-android/pull/2127)
4. Update the date in the message shown in the image below to September 1, 2020.

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
==========
Implements point 1 and 4 of #2539, this PR removes the Stats V4 availability banner and starts to activate the Stats V4 View if the API is available

Changelog
------------
| Description | Commit |
| ------------- | ------------- |
| Exclude the `MyStoreStatsAvailabilityCard` from the code base and UI since it will be no longer needed, removes from the `AppPrefs` all flags configuration related to the Stats Availability Banner  | 1c6f8b9430bf9934db77d42ab4c377b0d1c3c035 |
| Effectively starts to activate the Stats V4 right away if the V4 API is available. Removes from the `AppPrefs` and `MainActivity` any code related to the user enablement of the V4 feature and every listener method implementation and declaration related to it |  dddd849e26764f72be1c36e4455a74d849dd4e75 |
| Updates the `MyStoreStatsRevertedNoticeCard` message string, changing the Stats V3 due date warning from July 1 to September 1 | b0a673ccaed36c648fb968f551cf47e9ed8df90c |
| Rename the `isUsingV4Api` to `isV4StatsSupported` so the naming keeps the idea of the previous `isV4StatsUISupported` to ensure better code readability | b1dc8521ae79efab00df1a910b3064034b629e43 |
| Change the way the `MyStoreFragment` checks if the stats `TabLayout` is already inserted inside the `AppBar`, avoiding a bug of multiple tabs inserted simultaneously  | 9c386e8 |
| Add the removal of the `TabLayout` from the `MyStoreFragment` when the `onStop` event is called inside it, avoiding that the Tabs to still be visible even when the `MyStoreFragment` is no longer available | a7e940d | 
| The `MyStoreStatsRevertedNoticeCard` should now be displayed every time the Stats V3 is visible, so now the default value of `shouldDisplayV4StatsRevertedBanner` is `true`, so we will always present the banner and dismiss it only after the user hits the `dismiss` button and never see it again. | 4fc7992e5ebddd0f91fe6b42190bd461197e5d2b |

Screenshots when V4 API is only available
------------

| Before we just warned through a banner the Stats V4 availability and the option to try it | Now we change it right away without asking permission for the user |
| ------------- | ------------- |
| Screenshot_1592963863 | Screenshot_1592963801 |

| Before | After |
| ------------- | ------------- |
| Screenshot_1593188945 | Screenshot_1593189503 |

How to Test
------------
The best configuration to test this change is using the [WooCommerce Beta Tester Plugin](https://wordpress.org/plugins/woocommerce-beta-tester/) on your test site, this way the Woo Site version can be easily changed between V3 and V4, triggering the desirable scenario inside the Woo App.

### Scenario 1 ###
1. Log in with a store running V3 Woo and *does not have* WCAdmin installed.
2. On the My Store page make sure that the Stats V3 is showing up as the default with the appropriate "upgrade" card displayed, as shown in the `Before` screenshots above.
3. Install the WCAdmin plugin on the store and upgrade it to the V4 version.
4. Perform a pull to refresh.
5. Make sure that the Stats V4 are now been displayed and everything from the Stats V3 disappeared.

### Scenario 2 ###
1. Log in with a store running V4 Woo and *does have* WCAdmin installed.
2. Make sure that the Stats V4 are now been displayed with no warning banner, as shown in the `After` screenshots above.

Update release notes
------------

- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by AmandaRiu — CHANGES_REQUESTED** [review]
@ThomazFB Code chances look good! Tested through the app and noticed a couple issues:

**Double row of actionbar tabs issue:**
1. Log in with a store running v3 Woo and *does not* have WCAdmin installed.
2. On the My Store page the appropriate "upgrade" card is displayed. 
3. Install the WCAdmin plugin on the store.
4. Perform a pull to refresh. The card disappears and v4 stats are displayed, but there are no two tabbed rows:


**Actionbar tabs are not removed when downgrading to v3 stats**:
1. Log in with a store that has the WCAdmin plugin installed
2. Notice v4 stats are displayed
3. In wp-admin, disable the WCAdmin plugin. 
4. Notice the v4 toolbar tabs are still visible.


There should be some code in `MyStoreFragment` to add and remove the tabs from the toolbar: `removeTabLayoutFromAppBar(...)` - that may be a good place to start troubleshooting.

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -4,6 +4,9 @@
 * Updated the background color of the login views.
 * Moved privacy notice for CA users to the About App screen.
 * Disabled beta options and removed Improved Stats as selectable option from it
+* Set Stats V4 as the default choice
+* Remove Stats V4 Availability Banner Suggestion
```
[the remark this answers is not in the document]
Comment (ThomazFB) [reply RELEASE-NOTES.txt]: Hey @AmandaRiu, thanks for that information and suggestion, I'll try to leave the release notes concise as possible! I've committed the changes right here: 28aafbb412f4fd5e465bb7d3328f7d10420014f1

**Comment by ThomazFB** [comment]
Hey @AmandaRiu, thank you so much for your review, I've just pushed the changes around those two bugs you reported in these two commits:

Multiple tabs bug: 28aafbb412f4fd5e465bb7d3328f7d10420014f1
Tab not removed bug: a7e940d063257b060755bda3b51674fd27c5560d

Let me know what you think!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/SequenceExt.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+/**
+ * Extension function to verify if there's at least one instance of a given type inside a Sequence.
+ *
+ * Instead of trying to verify if there's an exact copy of a given element inside the sequence,
+ * this function create an alternative for the Sequence<T>.contains function checking only the instance type.
+ */
+inline fun <reified T> Sequence<View>.containsInstanceOf(element: T) =
+    filterIsInstance(T::class.java)
+        .toList()
+        .isNotEmpty()
```
Comment (AmandaRiu) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/SequenceExt.kt:14]: This is really slick 👍

**Review by AmandaRiu — CHANGES_REQUESTED** [review]
@ThomazFB Nice work 👍 both bugs are fixed, but now I have a new bug. The "Upgrade" card isn't getting displayed on first login with a store running less than 4.0:

1. Uninstall app and reinstall
2. Login with a store running 3.9.x
3. Note the My Store tab does not display the "Upgrade to keep seeing your stats" card.

**Comment by ThomazFB** [comment]
Hey @AmandaRiu, thanks for catching this issue! Here's what I decided to solve this.

Since we discussed on slack that the `MyStoreStatsRevertedNoticeCard` should now be displayed every time the Stats V3 is visible, at first I decided to remove everything around the `shouldDisplayV4StatsRevertedBanner` and let the banner always there, but later I started to notice that won't be a good idea because it would remove from the user the `dismiss` option to not see this banner anymore.

With that said, I decided to only change the default value of `shouldDisplayV4StatsRevertedBanner` to `true`, so we will always present the banner and dismiss it only after the user hits the `dismiss` button and never see it again.

Here's the commit: 4fc7992e5ebddd0f91fe6b42190bd461197e5d2b

Let me know what you think.

**Review by AmandaRiu — APPROVED** [review]
@ThomazFB Looks good! Nice work :) :shipit:

**Pull request merged**